src/main.rs

```diff
@@ -220,39 +218,61 @@ fn main() {
     game.renderer.camera.pos = cgmath::Point3::new(0.5, 13.2, 0.5);
 
     let mut events = sdl.event_pump().unwrap();
+    
+    let frame_rate = *game.console.lock().unwrap().get(settings::R_MAX_FPS);
+    let mut last_frame = time::now();
+
+    let frame_time = (time::Duration::seconds(1).num_nanoseconds().unwrap() as f64) / frame_rate as f64;
+    
+    let mut diff = time::Duration::nanoseconds(0);
+    
     while !game.should_close {
-        let version = {
-            let mut res = game.resource_manager.write().unwrap();
-            res.tick();
-            res.version()
-        };
-
-        let now = time::now();
-        let diff = now - last_frame;
-        last_frame = now;
         let delta = (diff.num_nanoseconds().unwrap() as f64) / frame_time;
-        let (width, height) = window.drawable_size();
-
-        game.tick(delta);
-        game.server.tick(&mut game.renderer, delta);
-
-        game.renderer.update_camera(width, height);
-        game.server.world.compute_render_list(&mut game.renderer);
-        game.chunk_builder.tick(&mut game.server.world, &mut game.renderer, version);
-
-        game.screen_sys.tick(delta, &mut game.renderer, &mut ui_container);
-        game.console
-            .lock()
-            .unwrap()
-            .tick(&mut ui_container, &mut game.renderer, delta, width as f64);
-        ui_container.tick(&mut game.renderer, delta, width as f64, height as f64);
-        game.renderer.tick(&mut game.server.world, delta, width, height);
-
-        window.gl_swap_window();
-
-        for event in events.poll_iter() {
-            handle_window_event(&window, &mut game, &mut ui_container, event)
+        
+        diff = time::Duration::span(||{
```

Comment:
`span` is deprecated in the stdlib version of time which I hope to be able to move to in the future once its all stable